### DISCUSSION TITLE
Include more complex examples or keep them basic?

### DISCUSSION THREAD

**Discussion by jhodapp** [Jim-Hodapp-Coaching/esp32-wroom-rp#57] [Q&A]
I want to briefly discuss what we all want to do in terms of including more complex examples that use this crate. For example, do we include an example that uses the BME280 crate + our crate to send sensor values to an Ambi server? Or do we keep this level of application in its own separate repo? And what's our general criteria that we want to use to decide whether an example belongs in this repo or in its own?

Here's my thoughts on this:
1. If something might or definitely will turn into its own interesting project, keep it out of this crate's examples area
2. If it will remain static and is not intended to become something interesting on its own but does demonstrate more complex usage of our crate, we include it in this repo.

**Comment by calebbourg** [comment]
I agree with the criteria you've proposed. 

I would add that we might consider the cost/benefit ratio of adding other components to our examples in terms of complexity.

 For example, adding the BME280 crate and necessary hardware, together, might add enough complexity in terms of setup and troubleshooting that the benefits of providing the example are outweighed. On the other hand adding another crate as a dependency in an example to showcase a scenario where we generate data (through some software means as opposed to more hardware) might be simple enough to be worth it.

I think this is even more important if we expect our users to be able to successfully run examples themselves.

**Reply by jhodapp** [reply]
Great, thanks for your thoughts Caleb. So then our criteria becomes:
1. If something might or definitely will turn into its own interesting project, keep it out of this crate's examples area
2. If it will remain static and is not intended to become something interesting on its own but does demonstrate more complex usage of our crate, we include it in this repo
3. If the maintenance cost/burden of adding it as an example to our project's repo is high, keep it separate, if it keeps a similar level to what exists today, include it in this repo
